### ISSUE TITLE
compaction will not remove clustered row tombstone even if it's covered by higher level tombstone

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylladb#1387]
In `mutation_partition::do_compact()` we have this check:

``` c++
        if (can_purge_tombstone(row.deleted_at())) {
            row.remove_tombstone();
        }
```

This will only remove row's tombstone when it expires, but not if it's covered by partition or range tombstones. This may result in larger sstables than necessary.

This problem doesn't apply to row cells, where we drop them if they're covered by tombstones (@ `row::compact_and_expire()`):

```
            if (cell.is_covered_by(tomb)) {
                erase = true;
            }
```

**Comment by meox** [comment]
any update?

**Comment by tgrabiec** [comment]
`mutation_partition::do_compact()` is no longer used during sstable compaction since c2f0ee9b5f1f58943aa2a0ad26263be95fb57c17, and this issue no longer applies to sstable compaction.

`do_compact()`, which still has the problem, is still used during reconciliation of query results on the coordinator side.

**Comment by bhalevy** [comment]
Looks like this was already fixed by 44bb9d495b46a067205e4d7a8a1c7b8ec0a80aec (or before).
@tgrabiec please confirm

**Comment by tgrabiec** [comment]
Fixed by https://github.com/scylladb/scylladb/commit/44bb9d495b46a067205e4d7a8a1c7b8ec0a80aec
